lib/docker.js

```diff
@@ -464,16 +464,17 @@ Docker.prototype.getEvents = function(opts, callback) {
  * @param  {String}   repoTag  Repository tag
  * @param  {Object}   opts     Options (optional)
  * @param  {Function} callback Callback
+ * @param  {Object}   auth     Authentication (optional)
  * @return {Object}            Image
  */
-Docker.prototype.pull = function(repoTag, opts, callback) {
+Docker.prototype.pull = function(repoTag, opts, callback, auth) {
```

Comment:
https://github.com/apocas/dockerode/blob/master/lib/image.js#L84 push suffers from the same problem.
So let's keep the same weird/emergency pattern in pull, at least push and pull will be alike.

In the next major (breaking change implied) we will fix this.